### ISSUE TITLE
The primary login method is still the password-based login, not OAuth2 via Eauth

### ISSUE THREAD

**Issue by rht** [inblockio/micro-PKC#99]
We need to completely remove the password-based login.

**Comment by rht** [comment]
I got this error when trying `$wgAuthManagerAutoConfig['primaryauth'] = []`.
![2021-12-13-233222_523x455_scrot](https://user-images.githubusercontent.com/395821/145933481-9cbfcaec-fc6a-40a8-b0c5-9180bd3c4cd2.png)

**Comment by it-spiderman** [comment]
Yes, this is exactly what i feared. Issue here is that MW allows you to replace default auth with something else, but not completely remove it, as this extension does. This extension works around MW auth, instead of integrating into it.
Another very quick solution would be to just block Special:Login. Its kind of hacky, but if its never going to be used....
For more proper solution (if even there is one), i would need to spend some time investigating

**Comment by rht** [comment]
The problem is that `Special:Login` is needed for when e.g. a user wants to set their email address, or other operation that requires extra authentication.

**Comment by it-spiderman** [comment]
But if you allow that, doesnt that break the "OAuth exclusive" logic? You would still need to have a local pass to be able to complete this additional authentication.
Why not just redirect Special:Login to Special:OAuthLogin (SP name probably not right)?
Also, it would require disabling API login completely, maybe just leaving bot login available, but as discussed with @FantasticoFox , all MW apis might be going away, so thats not a big deal.

**Comment by it-spiderman** [comment]
Try this
```
$GLOBALS['wgHooks']['SpecialPage_initList'] = function( &$list ) {
     $this->list['Userlogin'] = \OAuth2Client::class;
}
```
to see if it works, and fits your needs. If yes, this hook handler needs to be properly integrated into extension

**Comment by rht** [comment]
I got this error with that:
```
/index.php/Main_Page TypeError: Return value of MediaWiki\HookContainer\HookContainer::getLegacyHandlers() must be of the type array, null returned
```

**Comment by it-spiderman** [comment]
ah, sorry, you need `return true;` in this closure

**Comment by rht** [comment]
I added `return true;`, still got the same error message.

**Comment by it-spiderman** [comment]
ok, then let me actually investigate the code :)

**Comment by it-spiderman** [comment]
oh stupid me, forgot to add to the array of hook handlers, this is the correct code
```
$GLOBALS['wgHooks']['SpecialPage_initList'][] = function( &$list ) {
     $this->list['Userlogin'] = \OAuth2Client::class;
     return true;
}
```

**Comment by rht** [comment]
That's not quite correct either. This is my version so far:
```php
$GLOBALS['wgHooks']['SpecialPage_initList'][] = function( &$list ) {
     $list['Userlogin'] = [\OAuth2Client::class];
     return true;
};
```

And I got a new error:
`/index.php/Special:UserLogin InvalidArgumentException: Provided specification lacks both 'factory' and 'class' parameters.`

**Comment by FantasticoFox** [comment]
Wallet login is default login method and works.
